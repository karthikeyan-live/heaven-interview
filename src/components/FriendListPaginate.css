:local(.pageButtonList) {
    display: flex;
    justify-content: space-between;
    align-items: center;
}

:local(.btnAction), :local(.btnAction):active, :local(.btnAction):focus, :local(.btnAction):hover {
    margin: 2px;
    width: 40px;
    color: #5C75B0;
}