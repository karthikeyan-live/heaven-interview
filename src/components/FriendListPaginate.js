import React, { Component } from 'react';
import PropTypes from 'prop-types'; // to avoid the console warning
import styles from './FriendListPaginate.css';

class FriendListPaginate extends Component {
    render() {
        return (
            <div className={styles.pageButtonList}>
                <div>
                    <button className={`btn btn-default ${styles.btnAction}`}
                        disabled={this.props.activePage <= 1}
                        onClick={() => this.props.changeActivePage(1)}>
                        <i className="fa fa-fast-backward" />
                    </button>
                    <button className={`btn btn-default ${styles.btnAction}`}
                        disabled={this.props.activePage <= 1}
                        onClick={() => this.props.changeActivePage(this.props.activePage - 1)}>
                        <i className="fa fa-backward" />
                    </button>
                </div>
                <b>{this.totalPageCount() ? this.props.activePage : 0} of {this.totalPageCount()}</b>
                <div>
                    <button className={`btn btn-default ${styles.btnAction}`}
                        disabled={this.props.activePage >= this.totalPageCount()}
                        onClick={() => this.props.changeActivePage(this.props.activePage + 1)}>
                        <i className="fa fa-forward" />
                    </button>
                    <button className={`btn btn-default ${styles.btnAction}`}
                        disabled={this.props.activePage >= this.totalPageCount()}
                        onClick={() => { this.props.changeActivePage(this.totalPageCount()) }}>
                        <i className="fa fa-fast-forward" />
                    </button>
                </div>
            </div>
        );
    };

    componentDidUpdate(prevProps) {
        // if friend list have some item and no page is in view, then navigate to 1st page
        if (!this.props.activePage && this.totalPageCount()) {
            this.props.changeActivePage(1)
        }

        // if active page is greater than the total page count, navigate to the last page
        // helps when deleting items from the last page
        if (this.props.activePage > this.totalPageCount()) {
            this.props.changeActivePage(this.totalPageCount())
        }
    }

    totalPageCount = () => {
        return Math.ceil(this.props.totalItemsCount / this.props.itemsCountPerPage);
    }
}
FriendListPaginate.propTypes = {
    activePage: PropTypes.number.isRequired,
    itemsCountPerPage: PropTypes.number.isRequired,
    totalItemsCount: PropTypes.number.isRequired,
    changeActivePage: PropTypes.func.isRequired
};

export default FriendListPaginate;