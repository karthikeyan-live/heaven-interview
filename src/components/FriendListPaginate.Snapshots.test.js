import React, { Component } from 'react';
import renderer from 'react-test-renderer';
import FriendListPaginate from './FriendListPaginate';

function renderFriendListPaginate(args) {
    let defaultProps = {
        activePage: 1,
        itemsCountPerPage: 2,
        totalItemsCount: 3,
        changeActivePage: jest.fn()
    };
    const props = { ...defaultProps, ...args };
    return renderer.create(<FriendListPaginate {...props} />)
}
describe('friend list paginate - snapshot test', () => {
    it('with default props component is rendered', () => {
        const component = renderFriendListPaginate();
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });

    it('navigate to 2nd page and check if legend updated', () => {
        const component = renderFriendListPaginate({ activePage: 2 });
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
})