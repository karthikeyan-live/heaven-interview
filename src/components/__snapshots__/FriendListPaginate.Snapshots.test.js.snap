exports[`test navigate to 2nd page and check if legend updated 1`] = `
<div
  className={undefined}>
  <div>
    <button
      className="btn btn-default undefined"
      disabled={false}
      onClick={[Function]}>
      <i
        className="fa fa-fast-backward" />
    </button>
    <button
      className="btn btn-default undefined"
      disabled={false}
      onClick={[Function]}>
      <i
        className="fa fa-backward" />
    </button>
  </div>
  <b>
    2
     of 
    2
  </b>
  <div>
    <button
      className="btn btn-default undefined"
      disabled={true}
      onClick={[Function]}>
      <i
        className="fa fa-forward" />
    </button>
    <button
      className="btn btn-default undefined"
      disabled={true}
      onClick={[Function]}>
      <i
        className="fa fa-fast-forward" />
    </button>
  </div>
</div>
`;

exports[`test with default props component is rendered 1`] = `
<div
  className={undefined}>
  <div>
    <button
      className="btn btn-default undefined"
      disabled={true}
      onClick={[Function]}>
      <i
        className="fa fa-fast-backward" />
    </button>
    <button
      className="btn btn-default undefined"
      disabled={true}
      onClick={[Function]}>
      <i
        className="fa fa-backward" />
    </button>
  </div>
  <b>
    1
     of 
    2
  </b>
  <div>
    <button
      className="btn btn-default undefined"
      disabled={false}
      onClick={[Function]}>
      <i
        className="fa fa-forward" />
    </button>
    <button
      className="btn btn-default undefined"
      disabled={false}
      onClick={[Function]}>
      <i
        className="fa fa-fast-forward" />
    </button>
  </div>
</div>
`;
